aiida/calculations/transfer.py

```diff
@@ -0,0 +1,211 @@
+    new remote folder saved in a RemoteData node).
+
+    Only files from the local computer and a single remote computer can be moved with a single
+    instance of this CalcJob (so you can't directly move files from remote to remote and all
+    RemoteData nodes must be from the same machine).
+
+    The user needs to provide three inputs:
+    * `instructions`: a dict node specifying which files to copy from which nodes.
+    * `source_nodes`: a dict of nodes, each with a unique identifier label as its key.
+    * `metadata.computer`: the computer that contains the remote files and will contain the final
+    RemoteData node.
+
+    The `instructions` dict must have the `retrieve_files` flag. The CalcJob will create a new
+    folder in the remote machine (RemoteData) and put all the files there and will either:
+        (1) leave them there (`retrieve_files = False`) or ...
+        (2) create a local FolderData and copy there all the files (`retrieve_files = True`)
+
+    The `instructions` dict must also contain at least one list with specifications of which files
+    to copy and from where. All these lists take tupples of 3 that have the following format:
+
+        [ ..., ( source_node_label, path_to_file_in_source, path_to_file_in_target), ... ]
+
+    The two main lists to include are `local_files` (for files to be taken from FolderData nodes)
+    and `remote_files` (for files to be taken from RemoteData nodes). Alternatively, files inside
+    of RemoteData nodes can instead be put in the `symlink_files` list: the only difference is that
+    files from the first list will be fully copied in the target RemoteData folder, whereas for the
+    files in second list only a symlink to the original file will be created there. This will only
+    affect the content of the final RemoteData target folder, but in both cases the full file will
+    be copied back in the local target FolderData (if `retrieve_files = True).
+    """
+
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+
+        spec.input(
+            'instructions',
+            valid_type=orm.Dict,
+            help='A dictionary containing the `retrieve_files` flag and at least one of the file lists:'
+            '`local_files`, `remote_files` and/or `symlink_files`.',
+        )
+        spec.input_namespace(
+            'source_nodes',
+            valid_type=(orm.FolderData, orm.RemoteData),
+            dynamic=True,
+            help='All the nodes that contain files referenced in the instructions.',
+        )
+
+        # The transfer just needs a computer, the code are resources are set here
+        spec.inputs['code'].default = None
+        spec.inputs['code'].valid_type = (orm.Code, type(None))
+        spec.inputs['metadata']['computer'].required = True
+        spec.inputs['metadata']['options']['resources'].default = {
+            'num_machines': 1,
+            'num_mpiprocs_per_machine': 1,
+        }
```

Comment:
Well, you can check [the other PR](https://github.com/ramirezfranciscof/aiida-core/pull/8) in my fork where I already changed a bit the structure of the checks performed by the scheduler on the resources. Previously it had multiple raises in different conditional branches, but now it is all unified in one so it should be as easy as setting a flag that disables that single raise (currently `requires_resources`) or removing that raise altogether.

On the other hand, at the end of the process I still had to do some setting:
```
        if self.num_machines is None:
            self.num_machines = 1
        (...)
        if self.num_mpiprocs_per_machine is None:
            self.num_mpiprocs_per_machine = 1 
```
Because this is then used again in `presubmit` of the `CalcJob`. Line 382 ALWAYS tries to load the total number of processors:
```
subst_dict = {'tot_num_mpiprocs': job_tmpl.job_resource.get_tot_num_mpiprocs()}
for key, value in job_tmpl.job_resource.items():
      subst_dict[key] = value
mpi_args = [arg.format(**subst_dict) for arg in computer.get_mpirun_command()]
```

Yet this is only used if going with mpi (lines 414-419):
```
if this_withmpi:
    this_argv = (mpi_args + extra_mpirun_params + [this_code.get_execname()] +
                             (code_info.cmdline_params if code_info.cmdline_params is not None else []))
else:
    this_argv = [this_code.get_execname()] + (code_info.cmdline_params
                                                          if code_info.cmdline_params is not None else [])
```

I mean, if this is the last guard (which seems to be the case) it shouldn't be hard to change, I can add the modifications to the branch where I adapted the scheduler, I just wanted to check with you how to you if this is just putting the `subst_dict` parsing inside the "if mpi" clause or if perhaps we need to discuss something regarding how we conceptualize the meaning of this parameters (i.e. it could be say that even when not using mpi, whatever command you run is using at least some resources on the machine and thus you need at least one processor of at least one computer...).